superduperdb/base/config.py

```diff
@@ -4,25 +4,27 @@
 canot contain any other imports from this project.
 """
 
+import dataclasses as dc
 import json
 import os
 import typing as t
 from enum import Enum
 
-from .jsonable import Factory, JSONable
-
 _CONFIG_IMMUTABLE = True
 
 
-class BaseConfigJSONable(JSONable):
+@dc.dataclass
+class BaseConfigJSONable:
+    _lock: t.ClassVar[bool] = False
```

Comment:
What does this do?

Reply:
This is for locking the config after initialisation